library/azure_rm_appserviceplan.py

```diff
@@ -0,0 +1,354 @@
+        location: "eastus"
+        sku: S1
+        is_linux: true
+        number_of_workers: 1
+
+    - name: update sku of existing windows app service plan
+      azure_rm_appserviceplan:
+        name: "windowsplan2"
+        resource_group: "appserviceplan_rg"
+        location: "eastus"
+        sku: S2
+'''
+
+RETURN = '''
+azure_appserviceplan:
+    description: Facts about the current state of the app service plan
+    returned: always
+    type: dict
+    sample: {
+            "id": "/subscriptions/<subs_id>/resourceGroups/ansiblewebapp1_plan/providers/Microsoft.Web/serverfarms/win_appplan11"
+    }
+'''
+
+import time
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from msrest.serialization import Model
+    from azure.mgmt.web.models import (
+        app_service_plan, AppServicePlan, SkuDescription
+    )
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+def _normalize_sku(sku):
+    if sku is None:
+        return sku
+
+    sku = sku.upper()
+    if sku == 'FREE':
+        return 'F1'
+    elif sku == 'SHARED':
+        return 'D1'
+    return sku
+
+
+def get_sku_name(tier):
+    tier = tier.upper()
+    if tier == 'F1' or tier == "FREE":
+        return 'FREE'
+    elif tier == 'D1' or tier == "SHARED":
+        return 'SHARED'
+    elif tier in ['B1', 'B2', 'B3', 'BASIC']:
+        return 'BASIC'
+    elif tier in ['S1', 'S2', 'S3']:
+        return 'STANDARD'
+    elif tier in ['P1', 'P2', 'P3']:
+        return 'PREMIUM'
+    elif tier in ['P1V2', 'P2V2', 'P3V2']:
+        return 'PREMIUMV2'
+    else:
+        return None
+
+
+class AzureRMAppServicePlans(AzureRMModuleBase):
+    """Configuration class for an Azure RM App Service Plan resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str'
+            ),
+            sku=dict(
+                type='str'
+            ),
+            is_linux=dict(
+                type='bool',
+                default=False
+            ),
+            number_of_workers=dict(
+                type='str'
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+
+        self.sku = None
+        self.is_linux = None
+        self.number_of_workers = 1
+
+        self.tags = None
+
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_appserviceplan=None)
```

Comment:
ansible_facts?

Reply:
this is from upstream,  will fix upstream, then sync to role